pkg/mons/restore_quorum.go

```diff
@@ -0,0 +1,252 @@
+/*
+Copyright 2023 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mons
+
+import (
+	ctx "context"
+	"encoding/base64"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/rook/kubectl-rook-ceph/pkg/debug"
+	"github.com/rook/kubectl-rook-ceph/pkg/exec"
+	"github.com/rook/kubectl-rook-ceph/pkg/k8sutil"
+	log "github.com/sirupsen/logrus"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func RestoreQuorum(context *k8sutil.Context, operatorNamespace, clusterNamespace, goodMon string) {
+	err := restoreQuorum(context, operatorNamespace, clusterNamespace, goodMon)
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+}
+
+func restoreQuorum(context *k8sutil.Context, operatorNamespace, clusterNamespace, goodMon string) error {
+	monCm, err := context.Clientset.CoreV1().ConfigMaps(clusterNamespace).Get(ctx.TODO(), MonConfigMap, v1.GetOptions{})
+	if err != nil {
+		log.Fatalf("failed to get mon configmap %s %v", MonConfigMap, err)
+	}
+
+	monData := monCm.Data["data"]
+	monEndpoints := strings.Split(monData, ",")
+
+	var goodMonPublicIp, goodMonPort string
+	var badMons []string
+	for _, m := range monEndpoints {
+		monName, monEndpoint, _ := strings.Cut(m, "=")
+		if monName == goodMon {
+			goodMonPublicIp, goodMonPort, _ = strings.Cut(monEndpoint, ":")
```

Comment:
Handle if there is no `:`
```suggestion
			goodMonPublicIp, goodMonPort, ok = strings.Cut(monEndpoint, ":")
```